crawl-ref/source/initfile.cc

```diff
@@ -1632,7 +1637,8 @@ void read_init_file(bool runscript)
 
     if (f.error())
         return;
-    Options.read_options(f, runscript);
+    string option_file_contents = Options.read_options(f, runscript);
+    Options.file_contents += "\n" + option_file_contents + "\n\n";
```

Comment:
this will result in 2 copies of `option_file_contents` - one for the first `+`, and one for the `+=`. It would be better to append each element separately using `+=`. The other uses of `+` in this diff will similarly do some extra allocation you could avoid.

sorry for lurking, I just really like this project and sometimes take a peek at what y'all are up to :)

Reply:
Are you saying that there will be two copies of the options file in the dump? Because some simple testing reveals that is not the case... Otherwise I don't really understand your point.

Glad to hear you like the project.

Reply:
Just to clarify my understanding (which I believe to be correct, I'm no expert):
If options_file_contents contains one thing, eg `"test"`, then line 1641 creates the string `"\ntest\n\n"`, and appends this to `Options.file_contents`. I don't know where you think it is being duplicated.

Reply:
I believe @brianlheim is referring to temporary copies of the string containing the file contents, caused by the order of operations used: the string "\n" is created, then the file contents is copied onto the end of that, then "\n\n" is copied onto the end of that, and the result is then copied onto `Options.file_contents`; so the full contents of the file are copied twice. They are absolutely correct that this is less than maximally efficient, but this is somewhat less of a concern in this particular case.

Reply:
@aidanholm - that is correct. You're right this is not a huge concern in this particular instance, but being neglectful of temporary copies can lead to bad performance overall if the mistake is made repeatedly. And it is made here several times already: `content += str + "\n"` in the worst case may cause a reallocation when `"\n"` is appended to `str`, which is pessimistic when both are going to be copied to `content` in the end anyway.

Reply:
Right, that makes sense, thanks guys.

Reply:
Np! Sorry, I know it's kind of rude of me to interject out of the blue like this. I should say that in general I think you guys do a better job keeping this codebase clean than I probably would. :) It gives me hope that if/when I start to contribute it'll be fun and easy!

Reply:
If you think that's bad, just look at line 1798 of this diff :)